docker-compose.yml

```diff
@@ -78,102 +123,174 @@ services:
     image: opencti/connector-import-file-stix:6.0.10
+    deploy:
+      <<: *connector-deploy-settings
     environment:
-      - OPENCTI_URL=http://opencti:8080
-      - OPENCTI_TOKEN=${OPENCTI_ADMIN_TOKEN}
       - CONNECTOR_ID=${CONNECTOR_IMPORT_FILE_STIX_ID} # Valid UUIDv4
       - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
       - CONNECTOR_NAME=ImportFileStix
-      - CONNECTOR_VALIDATE_BEFORE_IMPORT=true # Validate any bundle before import
+      - CONNECTOR_VALIDATE_BEFORE_IMPORT=false # Validate any bundle before import
       - CONNECTOR_SCOPE=application/json,text/xml
       - CONNECTOR_AUTO=true # Enable/disable auto-import of file
-      - CONNECTOR_CONFIDENCE_LEVEL=15 # From 0 (Unknown) to 100 (Fully trusted)
-      - CONNECTOR_LOG_LEVEL=info
     restart: always
     depends_on:
       - opencti
+    networks:
+      - cfa_opencti_network
+
   connector-import-document:
+    <<: *connector-environment-vars
     image: opencti/connector-import-document:6.0.10
+    deploy:
+      <<: *connector-deploy-settings
     environment:
-      - OPENCTI_URL=http://opencti:8080
-      - OPENCTI_TOKEN=${OPENCTI_ADMIN_TOKEN}
       - CONNECTOR_ID=${CONNECTOR_IMPORT_DOCUMENT_ID} # Valid UUIDv4
       - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
       - CONNECTOR_NAME=ImportDocument
       - CONNECTOR_VALIDATE_BEFORE_IMPORT=true # Validate any bundle before import
       - CONNECTOR_SCOPE=application/pdf,text/plain,text/html
       - CONNECTOR_AUTO=true # Enable/disable auto-import of file
       - CONNECTOR_ONLY_CONTEXTUAL=false # Only extract data related to an entity (a report, a threat actor, etc.)
-      - CONNECTOR_CONFIDENCE_LEVEL=15 # From 0 (Unknown) to 100 (Fully trusted)
-      - CONNECTOR_LOG_LEVEL=info
-      - IMPORT_DOCUMENT_CREATE_INDICATOR=true
     restart: always
     depends_on:
       - opencti
+    networks:
+      - cfa_opencti_network
+
+  portainer-agent:
+    image: portainer/agent
+    environment:
+      # REQUIRED: Should be equal to the service name prefixed by "tasks." when
+      # deployed inside an overlay network
+      - AGENT_CLUSTER_ADDR=tasks.portainer-agent
+      # AGENT_PORT: 9001
+      # LOG_LEVEL: debug
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /var/lib/docker/volumes:/var/lib/docker/volumes
+    deploy:
+      mode: global
+    networks:
+      - cfa_opencti_network
+
+  portainer:
+    image: portainer/portainer
+    command: >
+      -H tcp://tasks.portainer-agent:9001 --tlsskipverify
+      --admin-password="$$2y$$05$$pZDaPit5nMFKZAOdT4m.v.NjuCJWzgK/1h4zkTbp1hNyuHdcvUhy6"
+    ports:
+      - "9010:9000"
+      - "8000:8000"
+    volumes:
+      - portainer_data:/data
+    deploy:
+      replicas: 1
+      placement:
+        constraints:
+          - node.role == manager
+    networks:
+      - cfa_opencti_network
+
 
 volumes:
-  esdata:
   s3data:
+    driver: local
+    driver_opts:
+      type: none
+      o: bind
+      device: /home/ubuntu/opencti/volumes/s3data
```

Comment:
Are we storing s3 data **_on_** the drive?

Reply:
We should be storing on s3, removing this after closing on the S3 issue